### PR TITLE
nixos/fwupd: Make daemon.conf structured

### PR DESCRIPTION
**Pull request by zhaofengli**
###### Description of changes

This enables users to add extra configurations to [/etc/fwupd/daemon.conf](https://github.com/fwupd/fwupd/blob/main/data/daemon.conf) (e.g., `DisabledDevices`, `IgnorePower`).

The local usecase is to allow installing custom unsigned updates (the Steam Deck BIOS repackaged for fwupd).

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/hardware/fwupd.nix [hunk 1]
```diff
@@ -120,6 +121,14 @@ in {
           Which fwupd package to use.
         '';
       };
+
+      extraDaemonConfigs = mkOption {
```
Comment (lovesegfault) [inline nixos/modules/services/hardware/fwupd.nix:125]: I think we commonly use singular, so
```suggestion
      extraDaemonConfig = mkOption {
```

?

**Comment by cab404** [comment]
Should we maybe use toml writer for this?

**Comment by zhaofengli** [comment]
> Should we maybe use toml writer for this?

Let's keep it simple for now. Appending our stuff to the end with the generator might not compose as well.

**Comment by jtojnar** [comment]
That’s what we are heading towards with [RFC 42](https://github.com/NixOS/rfcs/blob/fd559bdb8276901711e8813cbbf45a1f02578833/rfcs/0042-config-option.md). And it’s INI, or rather XDG/desktop/sytstemd keyfile type, not TOML.

**Comment by zhaofengli** [comment]
Ok, let's just generate the whole `daemon.conf` then.

**Comment by cab404** [comment]
> That’s what we are heading towards with [RFC 42](https://github.com/NixOS/rfcs/blob/fd559bdb8276901711e8813cbbf45a1f02578833/rfcs/0042-config-option.md). And it’s INI, or rather XDG/desktop/sytstemd keyfile type, not TOML.

I found some references to toml library in fwupd, thought it is)

**Review by RaitoBezarius — APPROVED** [review]
Looks good, can we have a release note on this?

**Review by lovesegfault — APPROVED** [review]

**Pull request merged**